Task5/script.js

```diff
@@ -0,0 +1,26 @@
+const app = document.getElementById("js-app");
+const fragment = document.createDocumentFragment();
+
+const data = [
+  { to: "bookmark.html", img: "1.png", alt: "画像1", text: "ブックマーク" },
+  { to: "message.html", img: "2.png", alt: "画像2", text: "メッセージ" }
+];
+
+const getData = new Promise((resolve) => resolve(data));
+
+getData.then(() => {
+  data.forEach((list) => {
+    const li = document.createElement("li");
+    const a = document.createElement("a");
+    const img = document.createElement("img");
+
+    a.href = `/${list.to}`;
+    a.textContent = list.text;
+    img.src = list.img;
+    img.alt = list.alt;
+
+    fragment.appendChild(li).appendChild(a).prepend(img);
+  });
```

Comment:
【IMO】
I received a value in resolve, I'll definitely use it!

```suggestion
getData.then((value) => {
//If we console.log(value) on this line
//[
//{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"},
//{to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
//]
//And then it comes out.Please check it out.
  value.forEach((list) => {
    const li = document.createElement("li");
    const a = document.createElement("a");
    const img = document.createElement("img");

    a.href = `/${list.to}`;
    a.textContent = list.text;
    img.src = list.img;
    img.alt = list.alt;

    fragment.appendChild(li).appendChild(a).prepend(img);
  });
```

Reply:
I was able to confirm that it also works as value.forEach instead of data.forEach by putting value as the argument of then!

I also understood that the array was output as console.log(value) as suggested, but I couldn't tell the difference because the behavior was the same even if the data was in the data state.

Can you please tell me the difference between using these values as arguments and the current situation?

Reply:
As a result, the same thing happens, so it's hard to understand. 
In the case of value, the resolved value is received in .then and that value is output. When data is forEach, it is in a state where it refers to const data instead of receiving the resolved value. So I suggested using value to mean receiving the resolved value in .then. 

By the way, the name of value can be anything. This time, value is used for clarity. For example, when using the rejected value in .catch, you can use error.

Reply:
Thank you sen for the explanation.

This is how I understood it
〇The value of the then argument -> Uses the value (data) received in resolve as the resolved value
〇In the case of data -> it is not using the resolved value, but simply reading an array

To use the value received from data in the Promise instance, you need to specify the value in the then argument.

Is that correct?